File: src/main/java/no/nav/sbl/dialogarena/mininnboks/provider/rest/resources/ResourcesController.java
package no.nav.sbl.dialogarena.mininnboks.provider.rest.resources;

import no.nav.sbl.dialogarena.mininnboks.consumer.TekstService;
import no.nav.sbl.dialogarena.mininnboks.provider.LinkService;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import java.util.HashMap;
import java.util.Map;

import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
import static no.nav.sbl.dialogarena.mininnboks.consumer.domain.Temagruppe.GODKJENTE_FOR_INNGAAENDE_SPORSMAAL;
import static org.springframework.util.StringUtils.collectionToDelimitedString;

@Path("/resources")
@Produces(APPLICATION_JSON + "; charset=UTF-8")
public class ResourcesController {

    @Inject
    TekstService tekstService;

    @GET
    public Map<String, Object> hentTekster() {
        Map<String, Object> tekster = new HashMap<>();

        tekster.putAll(tekstService.hentTekster());

        tekster.put("skriv.ny.link", LinkService.TEMAVELGER_LINK);
        tekster.put("brukerprofil.link", LinkService.BRUKERPROFIL_LINK);
        tekster.put("saksoversikt.link", LinkService.SAKSOVERSIKT_LINK);
        tekster.put("temagruppe.liste", collectionToDelimitedString(GODKJENTE_FOR_INNGAAENDE_SPORSMAAL, " "));

        return tekster;
    }

}

File: src/main/java/no/nav/sbl/dialogarena/mininnboks/provider/rest/resources/ResourcesController.kt
package no.nav.sbl.dialogarena.mininnboks.provider.rest.resources

import io.ktor.application.call
import io.ktor.response.respond
import io.ktor.routing.Route
import io.ktor.routing.get
import io.ktor.routing.route
import no.nav.sbl.dialogarena.mininnboks.consumer.TekstService
import no.nav.sbl.dialogarena.mininnboks.consumer.TekstServiceImpl
import no.nav.sbl.dialogarena.mininnboks.consumer.domain.Temagruppe
import no.nav.sbl.dialogarena.mininnboks.provider.LinkService

fun Route.ResourcesController() {

    route("/resources") {

        get {
            var tekstService: TekstService = TekstServiceImpl()

            val tekster: MutableMap<String?, Any?> = HashMap()
            tekster.putAll(tekstService!!.hentTekster()!!)
            tekster["skriv.ny.link"] = LinkService.TEMAVELGER_LINK
            tekster["brukerprofil.link"] = LinkService.BRUKERPROFIL_LINK
            tekster["saksoversikt.link"] = LinkService.SAKSOVERSIKT_LINK
            tekster["temagruppe.liste"] = Temagruppe.GODKJENTE_FOR_INNGAAENDE_SPORSMAAL.joinToString( " ")
            call.respond(tekster)
        }
    }
}
